src/field_theory/separable.lean

```diff
@@ -447,6 +448,16 @@ begin
   exact_mod_cast (enat.add_one_le_of_lt hq)
 end
 
+lemma square_free_of_separable {p : polynomial F} (hp : p ≠ 0)  (hsep : separable p) :
```

Comment:
Can you make this an iff?
Also, let's be consistent with `squarefree` instead of `square_free`.

Reply:
In general there are squarefree polynomial that are not separable, I guess the easiest example is `X ^ p - t` over `Fp(t)[X]`. The two notion are equivalent over perfect fields thought.

I changed `squarefree`.

Reply:
Ah yes, the iff depending on perfect fields can wait...